caput/containers/_util.py

```diff
@@ -1,28 +1,38 @@
 """Caput container utilities."""
 
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+
+    from ..mpiarray import SelectionLike
+    from ._core import ContainerBase
+
 import numpy as np
 
-from ..memdata import memh5
+from ..memdata import _memh5
 from ..mpiarray import _apply_sel
 from ._core import ContainerBase
 
 
-def empty_like(obj, **kwargs):
-    """Create an empty container like `obj`.
+def empty_like(obj: ContainerBase, **kwargs: dict) -> ContainerBase:  # noqa: D417
```

Comment:
I think this could benefit from a parameter name less-generic than `obj`, but I don't have a good suggestion.

Maybe `ref`, for "reference"?